src/psyclone/domain/lfric/kernel/lfric_kernel_metadata.py

```diff
@@ -0,0 +1,506 @@
+                    arg = InterGridArg.create_from_psyir(meta_arg)
+                elif vector_arg and not intergrid_arg:
+                    arg = FieldVectorArg.create_from_psyir(meta_arg)
+                else:
+                    arg = FieldArg.create_from_psyir(meta_arg)
+            else:
+                raise ParseError(
+                    f"Expected a 'meta_arg' entry to be a "
+                    f"field, a scalar or an operator, but found "
+                    f"'{meta_arg}'.")
+            kernel_metadata.meta_args.append(arg)
+
+        try:
+            meta_funcs = LFRicKernelMetadata._get_property(
+                spec_part, "meta_funcs")
+            args = walk(meta_funcs, Fortran2003.Ac_Value_List)
+        except ParseError:
+            meta_funcs = []
+
+        LFRicKernelMetadata.meta_reference_element = []
+        try:
+            LFRicKernelMetadata.meta_reference_element = \
+                LFRicKernelMetadata._get_property(
+                    spec_part, "meta_reference_element")
+        except ParseError:
+            pass
+        args = walk(LFRicKernelMetadata.meta_reference_element,
+                    Fortran2003.Ac_Value_List)
+        if not args:
+            LFRicKernelMetadata.meta_reference_element = []
+
+        # meta_mesh contains arguments which have
+        # properties.
+        try:
+            LFRicKernelMetadata.meta_mesh = LFRicKernelMetadata._get_property(
+                spec_part, "meta_mesh")
+        except ParseError:
+            # meta_mesh is not specified in the metadata
+            LFRicKernelMetadata.meta_mesh = []
+        return kernel_metadata
+
+    def lower_to_psyir(self):
+        '''Lower the metadata to language-level PSyIR.
+
+        :returns: metadata as stored in language-level PSyIR.
+        :rtype: :py:class:`psyclone.psyir.symbols.DataTypeSymbol`
+
+        '''
+        return DataTypeSymbol(
+            str(self.name), UnknownFortranType(self.fortran_string()))
+
+    @staticmethod
+    def _get_property(spec_part, property_name):
+        '''Internal utility that gets the property 'property_name' from an
+        fparser2 tree capturing LFRic metadata. It is assumed that
+        the code property is part of a type bound procedure and that
+        the other properties are part of the data declarations.
+
+        :param spec_part: the fparser2 parse tree containing the metadata.
+        :type spec_part: :py:class:`fparser.two.Fortran2003.Derived_Type_Def`
+        :param str property_name: the name of the property whose value \
+            is being extracted from the metadata.
+
+        :returns: the value of the property.
+        :rtype: :py:class:`fparser.two.Fortran2003.Name | \
+            :py:class:`fparser.two.Fortran2003.Array_Constructor`
+
+        :raises ParseError: if the metadata is invalid.
+
+        '''
+        if property_name.lower() == "code":
+            # The value of 'code' should be found in a type bound
```

Comment:
What happens if there's an interface (for multi-precision kernels)?

Reply:
What we do is add it to the list of things that we don't yet support :-)

Reply:
I've made a note in issue #1879, so this won't be forgotten.